go/vrf/ecvrf.go

```diff
@@ -43,3 +44,29 @@ func NewKey(curve elliptic.Curve, sk []byte) *PrivateKey {
 		d:         new(big.Int).SetBytes(sk),           // Use SK to derive the VRF secret scalar x
 	}
 }
+
+// ECVRFParams holds shared values across ECVRF implementations.
+// ECVRFParams also has generic algorithms that rely on ECVRFAux for specific sub algorithms.
+type ECVRFParams struct {
+	suiteString []byte         // Single nonzero octet specifying the ECVRF ciphersuite
+	ec          elliptic.Curve // Elliptic curve defined over F
+	n           int            // 2n is the length, in bytes, of a field element in F
```

Comment:
Maybe it's better to rename this to fieldLen which would be more descriptive and consistent with other parameters which define octet lengths (and pass the field length in P256SHA256TAI)? I don't think it's a big issue, but might still cause confusion as the IETF document defines 2n as the fixed parameter for a scheme.

Reply:
That's a good idea. Perhaps you could also weigh in on https://github.com/google/draft-irtf-cfrg-vrf/pull/5#discussion_r436810281

`ECVRF_hash_points` seems to emit an integer in the range `[0,fieldLen/2 == n]`. Does that make sense to you?

Reply:
It seems a bit unintuitive at first why it's okay to have only fieldLen/2 bits for this and I was also wondering :-). I briefly checked the proof in the paper https://eprint.iacr.org/2017/099.pdf, and it only relies on `ECVRF_hash_points` to give a random output, as the best the attacker can do to break the properties of the VRF (through exploiting `ECVRF_hash_points) is to hope to match a specific value of size fieldLen/2. This matches the overall security level (2^128), so it's fine.

Reply:
@reyzin - can you weigh in here?